sw/device/tests/alert_handler_lpg_sleep_mode_pings.c

```diff
@@ -0,0 +1,531 @@
+      kDifAlertHandlerLocalAlertAlertPingFail,
+      kDifAlertHandlerLocalAlertAlertIntegrityFail,
+      kDifAlertHandlerLocalAlertBusIntegrityFail,
+      kDifAlertHandlerLocalAlertEscalationIntegrityFail,
+      kDifAlertHandlerLocalAlertEscalationPingFail,
+      kDifAlertHandlerLocalAlertShadowedStorageError,
+      kDifAlertHandlerLocalAlertShadowedUpdateError};
+  dif_alert_handler_class_t loc_alert_classes[] = {
+      kDifAlertHandlerClassB, kDifAlertHandlerClassA, kDifAlertHandlerClassA,
+      kDifAlertHandlerClassA, kDifAlertHandlerClassA, kDifAlertHandlerClassA,
+      kDifAlertHandlerClassA};
+
+  dif_alert_handler_escalation_phase_t esc_phases[] = {
+      {.phase = kDifAlertHandlerClassStatePhase0,
+       .signal = 0,
+       .duration_cycles = 2000}};
+
+  dif_alert_handler_class_config_t class_config = {
+      .auto_lock_accumulation_counter = kDifToggleDisabled,
+      .accumulator_threshold = 0,
+      .irq_deadline_cycles = 10000,
+      .escalation_phases = esc_phases,
+      .escalation_phases_len = ARRAYSIZE(esc_phases),
+      .crashdump_escalation_phase = kDifAlertHandlerClassStatePhase1,
+  };
+
+  dif_alert_handler_class_config_t class_configs[] = {class_config,
+                                                      class_config};
+
+  dif_alert_handler_class_t classes[] = {kDifAlertHandlerClassA,
+                                         kDifAlertHandlerClassB};
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = ARRAYSIZE(alerts),
+      .local_alerts = loc_alerts,
+      .local_alert_classes = loc_alert_classes,
+      .local_alerts_len = ARRAYSIZE(loc_alerts),
+      .classes = classes,
+      .class_configs = class_configs,
+      .classes_len = ARRAYSIZE(class_configs),
+      .ping_timeout = ping_timeout,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+  // Enables alert handler irq.
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassb, kDifToggleEnabled));
+}
+
+/**
+ * Clear all alert_cause and local_alert_cause registers
+ */
+static void alert_handler_clear_cause_regs(void) {
+  bool is_cause;
+  // Loop over all alert_cause regs
+  for (int jj = 0; jj < ALERT_HANDLER_PARAM_N_ALERTS; jj++) {
+    CHECK_DIF_OK(dif_alert_handler_alert_acknowledge(&alert_handler, jj));
+  }
+
+  // Loop over all loc_alert_cause regs
+  for (int jj = 0; jj < 7; jj++) {
+    CHECK_DIF_OK(dif_alert_handler_local_alert_acknowledge(&alert_handler, jj));
+  }
+}
+
+/**
+ * Counts and returns the number of fired alerts
+ */
+static uint16_t alert_handler_num_fired_alerts(void) {
+  bool is_cause;
+  // Indicates if the ping_timeout_fail fired is fired.
+  bool is_cause_ping_timeout_fail;
```

Comment:
is this variable used somewhere?

Reply:
Thanks for catching this. I removed it :)